app/controllers/api/v3/provider_suggestions_controller.rb

```diff
@@ -7,7 +7,12 @@ def index
         return render(status: :bad_request) if params[:query].nil? || params[:query].length < 3
         return render(status: :bad_request) unless begins_with_alphanumeric(params[:query])
 
+        providers_ids = @recruitment_cycle.courses.findable.distinct.pluck(:provider_id)
```

Comment:
This logic shouldn't be here in the controller. Can we move it to a scope on `Provider`? Maybe `with_courses` or `with_active_courses` or something like that. Then the logic for `active_courses` should be on `Course` and the `Provider` scope would merge that in. They could both be easily tested in a bit more isolation.